src/stirling/utils/linux_headers.cc

```diff
@@ -627,9 +620,30 @@ Status InstallPackagedLinuxHeaders(const std::filesystem::path& lib_modules_dir)
   PX_ASSIGN_OR_RETURN(PackagedLinuxHeadersSpec packaged_headers,
                       FindClosestPackagedLinuxHeaders(kPackagedHeadersRoot, kernel_version));
   LOG(INFO) << absl::Substitute("Using packaged header: $0", packaged_headers.path.string());
-  PX_RETURN_IF_ERROR(ExtractPackagedHeaders(&packaged_headers));
+  const std::string version =
+      absl::Substitute("$0.$1.$2-pl", packaged_headers.version.version,
+                       packaged_headers.version.major_rev, packaged_headers.version.minor_rev);
+  const std::string staging_directory = absl::StrCat("/usr/src/staging/linux-headers-", version);
+  const std::string expected_directory = absl::StrCat("/usr/src/linux-headers-" + version);
+  // Verify that the target directory doesn't already exist.
+  // If someone built a tar.gz with an incorrect directory structure, this check wouldn't save us.
+  if (fs::Exists(expected_directory)) {
+    return error::Internal(
+        "Not attempting to install packaged headers because the target directory already exists: "
+        "$0",
+        expected_directory);
+  }
+  // Extract the packaged headers to a staging directory, stripping the expected target directory
+  // prefix.
+  PX_RETURN_IF_ERROR(
+      ExtractPackagedHeaders(&packaged_headers, staging_directory, expected_directory));
```

Comment:
It seems like there is a side effect in `ExtractPackagedHeaders`. I wonder if we can pass in the "headers package path" and the "staging dir" as separate and immutable args., and then below here just use `staging_directory` on lines 640, 642, and 645.

Reply:
Great point! I have passed  `staging_directory` around instead of mutating `packaged_headers.path`. Does that address your comment?